vespalog/src/main/java/com/yahoo/log/LogMessage.java

```diff
@@ -85,19 +109,37 @@ public static LogMessage parseNativeFormat(String msg) throws InvalidLogFormatEx
         }
 
         Level msgLevel = LogLevel.parse(m.group(6));
-        Long timestamp = parseTimestamp(m.group(1));
+        Instant timestamp = parseTimestamp(m.group(1));
+        String threadProcess = m.group(3);
 
-        return new LogMessage(m.group(1), timestamp, m.group(2), m.group(3),
+        return new LogMessage(timestamp, m.group(2), parseProcessId(threadProcess), parseThreadId(threadProcess),
                               m.group(4), m.group(5), msgLevel,
                               m.group(7));
     }
 
-    private static long parseTimestamp(String timeStr) throws InvalidLogFormatException {
+    private static Instant parseTimestamp(String timeStr) throws InvalidLogFormatException {
         try {
-            return (long) (Double.parseDouble(timeStr) * 1000);
+            long nanoseconds = (long) (Double.parseDouble(timeStr) * 1_000_000_000L);
+            return Instant.ofEpochSecond(0, nanoseconds);
         } catch (NumberFormatException e) {
-            throw new InvalidLogFormatException("Invalid time string:" + timeStr);
+            throw new InvalidLogFormatException("Invalid time string: " + timeStr);
+        }
+    }
+
+    private static long parseProcessId(String threadProcess) {
+        int slashIndex = threadProcess.indexOf('/');
+        if (slashIndex == -1) {
+            return Long.parseLong(threadProcess);
+        }
+        return Long.parseLong(threadProcess.substring(0, slashIndex));
+    }
+
+    private static long parseThreadId(String threadProcess) {
+        int slashIndex = threadProcess.indexOf('/');
+        if (slashIndex == -1) {
+            return -1;
```

Comment:
the format method thinks "0" means unknown but here it's "-1" that means unknown